doc/user-guide/dask.rst

```diff
@@ -202,125 +158,73 @@ NumPy array:
+.. _dask.chunks:
 
-NumPy ufuncs like ``np.sin`` transparently work on all xarray objects, including those
-that store lazy Dask arrays:
-
-.. ipython:: python
-
-    import numpy as np
+Chunking and performance
+~~~~~~~~~~~~~~~~~~~~~~~~
 
-    np.sin(rechunked)
+The way a dataset is chunked can be critical to performance when working with large datasets. You'll want have chunk sizes large enough to reduce the number of chunks that Dask has to think about (to reduce overhead from the task graph) but also small enough so that many of them can fit in memory at once.
 
-To access Dask arrays directly, use the
-:py:attr:`DataArray.data <xarray.DataArray.data>` attribute. This attribute exposes
-array data either as a Dask array or as a NumPy array, depending on whether it has been
-loaded into Dask or not:
+.. tip::
 
-.. ipython:: python
+    A good rule of thumb is to create arrays with a minimum chunk size of at least one million elements (e.g., a 1000x1000 matrix). With large arrays (10+ GB), you may need larger chunks. See `Choosing good chunk sizes in Dask <https://blog.dask.org/2021/11/02/choosing-dask-chunk-sizes?utm_source=xarray-docs>`_.
 
-    ds.temperature.data
+It can be helpful to choose chunk sizes based on your downstream analyses and to chunk as early as possible. Datasets with smaller chunks along the time axis, for example, can make time domain problems easier to parallelize since Dask can perform the same operation on each time chunk. If you're working with a large dataset with chunks that make downstream analyses challenging, you may need to rechunk your data. This is an expensive operation though, so is only recommended when needed.
 
-.. note::
+You can rechunk a dataset by:
 
-    ``.data`` is also used to expose other "computable" array backends beyond Dask and
-    NumPy (e.g. sparse and pint arrays).
+- Specifying ``chunks={}`` when reading in your dataset. If you know you'll want to do some spatial subsetting, for example, you could use ``chunks={'latitude': 10, 'longitude': 10}`` to specify small chunks across space. This can avoid loading subsets of data that span multiple chunks, thus reducing the number of file reads.
+- Using :py:meth:`Dataset.chunk` after you've already read in your dataset. For time domain problems, for example, you can use ``ds.chunk(time=TimeResampler())`` to rechunk according to a specified unit of time. ``ds.chunk(time=TimeResampler("MS"))``, for example, will set the chunks so that a month of data is contained in one chunk.
 
 .. _dask.automatic-parallelization:
 
-Automatic parallelization with ``apply_ufunc`` and ``map_blocks``
------------------------------------------------------------------
-
-.. tip::
-
-   Some problems can become embarrassingly parallel and thus easy to parallelize
-   automatically by rechunking to a frequency, e.g. ``ds.chunk(time=TimeResampler("YE"))``.
-   See :py:meth:`Dataset.chunk` for more.
+Parallelize custom functions with ``apply_ufunc`` and ``map_blocks``
```

Comment:
I didn't really touch this section because I'm not all that familiar with how people use `apply_ufunc` and `map_blocks`.